application/server/configuration.go

```diff
@@ -344,10 +344,22 @@ func updateSnykCodeQuality(settings lsp.Settings) {
 }
 
 // TODO store in config, move parsing to CLI
-func updatePath(settings lsp.Settings) {
-	err := os.Setenv("PATH", cachedOriginalPath+string(os.PathListSeparator)+settings.Path)
-	if err != nil {
-		log.Err(err).Msgf("couldn't add path %s", settings.Path)
+func updatePathFromSettings(settings lsp.Settings) {
+	// when changing the path from settings, we cache the original path first, to be able to restore it later
+	if len(cachedOriginalPath) == 0 {
+		cachedOriginalPath = os.Getenv("PATH")
+	}
+
+	if len(settings.Path) > 0 {
+		os.Unsetenv("Path") // unset the path first to work around issues on Windows OS, where PATH can be Path
+		err := os.Setenv("PATH", settings.Path+string(os.PathListSeparator)+cachedOriginalPath)
```

Comment:
somewhere there's a global var that provides a PathListSeparator string - maybe in config.go?